spinnaker-monitoring-daemon/tests/datadog_service_test.py

```diff
@@ -18,11 +18,49 @@
 import unittest
 import tempfile
 from mock import patch
-from StringIO import StringIO
 
 import datadog_service
-from datadog_service import DatadogMetricsService
-from ConfigParser import ConfigParser
+from datadog_service import DatadogMetricsService, DatadogArgumentsGenerator
+
+
+def service_generation_helper(config_data=[], datadog_options={},
+                              spinnaker_monitoring_options={}):
+    """
+    This utility lets you build a DatadogMetricsService in a DRY fashion
+    for all the tests below. You can inject additional parameters anywhere.
+
+    It guarantees the base invariants - api_key present,
+    'datadog' and 'dd_agent_config' keys exist in required arguments,
+    and that 'dd_agent_config' is correctly parsed as a file. These are
+    guarded against within the DatadogService.
+    """
+
+    data = ["[Main]", "api_key: FOUND_KEY"] + config_data
+    options = {'datadog': datadog_options, 'dd_agent_config': ''}
+    options.update(spinnaker_monitoring_options)
+    with tempfile.NamedTemporaryFile() as config:
+      config.write('\n'.join(data))
+      config.flush()
+      options['dd_agent_config'] = config.name
+      service = datadog_service.make_datadog_service(options)
+
+    return service
+
+def arguments_generator_helper(config_data=[], options={}, set_dd_agent_conf=True):
+    """
+    This utility lets you build a DatadogArgumentsGenerator in a DRY fashion
+    for all the tests below. You can inject additional parameters anywhere.
+    """
+
+    data = config_data
+    with tempfile.NamedTemporaryFile() as config:
+      config.write('\n'.join(data))
+      config.flush()
+      if set_dd_agent_conf:
+        options['dd_agent_cofig'] = config.name
```

Comment:
config spelling

Reply:
I thought I had another comment here -- not sure why I removed it.
I'd said that you dont really need this for the test since you arent testing the use of this. You test to see the exception if it isnt here which is still ok. You could test that it is picked up from teh file if it is there, but I think you already have that covered through the tests on the tags.

So you can remove this dd_agent_config attribute and teh set parameter for it, or add a test where you actually use it. Your call.